src/verifiers/identity/builder.ts

```diff
@@ -0,0 +1,142 @@
+import { v2, v3, WrappedDocument, utils, getData } from "@govtechsg/open-attestation";
+import {
+  VerificationFragmentType,
+  Verifier,
+  VerificationFragmentStatus,
+  SkippedVerificationFragment,
+  VerificationManagerOptions,
+} from "../../types/core";
+import { Reason } from "../../types/error";
+import { OpenAttestationIssuerIdentityVerifierCode } from "../../types/error";
+
+const name = "OpenAttestationIssuerIdentityVerifier";
+const type: VerificationFragmentType = "ISSUER_IDENTITY";
+type VerifierType = Verifier<WrappedDocument<v2.OpenAttestationDocument> | WrappedDocument<v3.OpenAttestationDocument>>;
+
+export interface VerifierResults<T = any> {
+  verifier: string; // Which verifier returned this result
+  identifier?: string; // Human readable interpretation of the issuer
+  status: VerificationFragmentStatus; // Status of this verification result
+  data?: T; // Other metadata returned by verifier
+  reason?: Reason; // Reasons for status
+}
+
+export type IssuerIdentityVerifier<
+  Document = WrappedDocument<v3.OpenAttestationDocument> | WrappedDocument<v2.OpenAttestationDocument>,
+  ResultData = any
+> = ({
+  document,
+  options,
+  issuerIndex,
+}: {
+  document: Document;
+  options: VerificationManagerOptions;
+  issuerIndex?: number;
+}) => Promise<VerifierResults<ResultData>>;
+
+export type IssuerIdentityVerifierDefinition = {
+  type: string;
+  verify: IssuerIdentityVerifier;
+};
+
+const skipFragment: SkippedVerificationFragment = {
+  status: "SKIPPED",
+  type,
+  name,
+  reason: {
+    code: OpenAttestationIssuerIdentityVerifierCode.SKIPPED,
+    codeString: OpenAttestationIssuerIdentityVerifierCode[OpenAttestationIssuerIdentityVerifierCode.SKIPPED],
+    message: `Verification of issuers' identity skipped`,
+  },
+};
+
+const defaultVerifier: IssuerIdentityVerifier = async ({ issuerIndex }) => {
+  return {
+    verifier: "SKIPPED_VERIFIER",
+    status: "SKIPPED",
+    reason: {
+      code: OpenAttestationIssuerIdentityVerifierCode.SKIPPED,
+      codeString: OpenAttestationIssuerIdentityVerifierCode[OpenAttestationIssuerIdentityVerifierCode.SKIPPED],
+      message: `No verifier found for issuer ${issuerIndex}`,
+    },
+  };
+};
+
+export const issuerIdentityVerifierBuilder = (verifiers: IssuerIdentityVerifierDefinition[]): VerifierType => {
+  const registeredVerifiers: { [identityProofType: string]: IssuerIdentityVerifier } = {};
+  verifiers.forEach((verifier) => (registeredVerifiers[verifier.type] = verifier.verify));
```

Comment:
if dev is idiot and no verifier provided?

Reply:
This won't be the final interface for dev, there will be a recommended default set. But let's say that dev is an idiot, it will have the default verifier that skips everything and then fails.